packages/legend-extension-dsl-data-space/src/stores/query/DataSpaceTemplateQueryPromotionReviewerStore.ts

```diff
@@ -0,0 +1,666 @@
+      (ws) => ws.workspaceId === this.workspaceName,
+    );
+  }
+
+  get isTemplateQueryIdValid(): boolean {
+    return !this.dataSpace?.executables
+      ?.filter(filterByType(DataSpaceExecutableTemplate))
+      .map((et) => et.id)
+      .includes(this.templateQueryId);
+  }
+
+  *initialize(
+    queryId: string | undefined,
+    dataSpacePath: string,
+  ): GeneratorFn<void> {
+    if (!this.initState.isInInitialState) {
+      return;
+    }
+    try {
+      this.initState.inProgress();
+      yield this.graphManagerState.graphManager.initialize(
+        {
+          env: this.applicationStore.config.env,
+          tabSize: DEFAULT_TAB_SIZE,
+          clientConfig: {
+            baseUrl: this.applicationStore.config.engineServerUrl,
+            queryBaseUrl: this.applicationStore.config.engineQueryServerUrl,
+            enableCompression: true,
+          },
+        },
+        {
+          tracerService: this.applicationStore.tracerService,
+        },
+      );
+      yield this.graphManagerState.initializeSystem();
+      // reset
+      this.graphManagerState.resetGraph();
+      // build dependencies
+      const dependencyManager =
+        this.graphManagerState.graphManager.createDependencyManager();
+      this.graphManagerState.graph.dependencyManager = dependencyManager;
+      yield this.graphManagerState.graphManager.buildDependencies(
+        this.graphManagerState.coreModel,
+        this.graphManagerState.systemModel,
+        dependencyManager,
+        new Map(),
+        this.graphManagerState.dependenciesBuildState,
+        {},
+      );
+      this.dataSpacePath = dataSpacePath;
+      if (queryId) {
+        let query: LightQuery | undefined;
+        try {
+          query = (yield this.graphManagerState.graphManager.getLightQuery(
+            queryId,
+          )) as LightQuery;
+        } catch {
+          query = undefined;
+        }
+        if (query) {
+          yield flowResult(this.loadQuery(query));
+        } else {
+          this.applicationStore.navigationService.navigator.updateCurrentLocation(
+            generateDataSpaceQuerySetupRoute(),
+          );
+        }
+      }
+      if (this.currentQuery) {
+        this.currentQueryProject = StoreProjectData.serialization.fromJson(
+          (yield this.depotServerClient.getProject(
+            this.currentQuery.groupId,
+            this.currentQuery.artifactId,
+          )) as PlainObject<StoreProjectData>,
+        );
+        const projectData = (yield Promise.all([
+          this.depotServerClient.getVersionEntities(
+            this.currentQuery.groupId,
+            this.currentQuery.artifactId,
+            this.currentQuery.versionId,
+          ),
+          this.sdlcServerClient.getConfiguration(
+            this.currentQueryProject.projectId,
+            undefined,
+          ),
+        ])) as [Entity[], PlainObject<ProjectConfiguration>];
+
+        const [currentProjectEntities, currentProjectConfiguration] = [
```

Comment:
I don't think we should build the full graph here. Even the system graph, unless we have a reason?
We should maybe create a light Dataspace and jsut buidl that from the Entity. 
So fetch the entity directly. 
We will do the compiler checks once they click procceed.

Reply:
I did try to build light graph at first, however, light graph won't generate existing dataspace's executables  e.g. 
<img width="923" alt="Screenshot 2024-04-15 at 9 39 42 AM" src="https://github.com/finos/legend-studio/assets/73408381/722a8cc7-bae5-41ae-9a63-fc825f1d92aa">

Then it has to append newly added executable entity plain object to original dataspace entity